monitoring/README.md

```diff
@@ -0,0 +1,185 @@
+# Monitoring
+
+The monitoring package is built on [OpenTelemetry](https://opentelemetry.io/) and provides implementations for creating custom spans, sending custom metrics, and propagating Trace IDs across HTTP requests, GCP Pub/Sub clients, and RabbitMQ clients.
+
+1. [Traces](#traces)
+    - [Use of the Tracer](#use-of-the-tracer)
+2. [Spans](#spans)
+    - [Automatic Correlation](#automatic-correlation)
+3. [Distributed Tracing](#distributed-tracing)
+    - [HTTP Tracing](#http-tracing)
+    - [PubSub Tracing](#pubsub-tracing)
+    - [RabbitMQ Tracing](#rabbitmq-tracing)
+    - [Middleware](#middleware)
+4. [Metrics](#metrics)
+    - [Use of the Metrics](#use-of-the-metrics)
+5. [Kubernetes Setup](#kubernetes-setup)
+6. [Examples](#examples)
+7. [Otel Documentation](#otel-documentation)
+
+## Traces
+
+The path of a request through your application.
+
+**Traces** give us the big picture of what happens when a request is made to an application. Whether your application is a monolith with a single database or a sophisticated mesh of services, traces are essential to understanding the full “path” a request takes in your application.
+
+### Use of the Tracer
+
+The library let's you initialise a Tracer. A Tracer has a name. In our case, the name is the service name (the APM name). Using the default Tracer, you can create Spans.
+
+Before you initialise the Tracer, first see the [Kubernetes Setup](#kubernetes-setup).
+Also, you can find examples of the Traces in the [examples](#examples).
+
+## Spans
+
+A **Span** represents a unit of work or operation. Spans are the building blocks of Trace. In OpenTelemetry, they include the following information:
+
+- Name
+- Parent span ID (empty for root spans)
+- Start and End Timestamps
+- [Span Context](https://opentelemetry.io/docs/concepts/signals/traces/#span-context)
+- [Attributes](https://opentelemetry.io/docs/concepts/signals/traces/#attributes)
+- [Span Events](https://opentelemetry.io/docs/concepts/signals/traces/#span-events)
+- [Span Links](https://opentelemetry.io/docs/concepts/signals/traces/#span-links)
+- [Span Status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status)
+
+Spans can be nested, as is implied by the presence of a parent span ID: child spans represent sub-operations. This allows spans to more accurately capture the work done in an application.
+
+Spans follow the [Open Telemetry Semantic Conventions](https://opentelemetry.io/docs/concepts/semantic-conventions/). That ensures correlation on naming between spans and [logs](../logger/).
+
+The naming conversions can be found in [internal/config/commontags.go](../internal/config/commontags.go).
+
+Creating a Span it requires a Tracer. The library provides a simple way to create Spans. Furthermore, exposing the Span interface you can easily add Attributes, Events, Links, Status and Record Errors in your Spans.
```

Comment:
`Furthermore, the exposed Span interface allows you to easily add a Status, Attributes, Events, Links to a Span and Record Errors in a Span.`